src/BrowserInterop/Extensions/CallBackInteropWrapper.cs

```diff
@@ -86,15 +114,31 @@ private CallBackInteropWrapper()
         public static CallBackInteropWrapper Create(Func<ValueTask> callback, object serializationSpec = null,
             bool getJsObjectRef = false)
         {
-            var res = new CallBackInteropWrapper
+            var dotnetCallback = DotNetObjectReference.Create(new JsInteropActionWrapper(callback));
+            var res = new CallBackInteropWrapper(dotnetCallback)
             {
-                CallbackRef = DotNetObjectReference.Create(new JsInteropActionWrapper(callback)),
                 SerializationSpec = serializationSpec,
                 GetJsObjectRef = getJsObjectRef
             };
             return res;
         }
 
-        public object CallbackRef { get; set; }
+        public void Dispose()
```

Comment:
is it because there was a leak and the callbackref was never cleaned up?

Reply:
yes it wasn't made visible that there's a disposable resource under that wrapper (the DotnetCallback) . By adding the IDisposable I just made it visible so that the user is aware of it.